src/lib/Libwin/passwd.c

```diff
@@ -4814,20 +4815,21 @@ logon_pw(char *username,
 	size_t credl,
 	int (*decrypt_func)(char *, int, size_t, char **),
 	int use_winsta,
-	char msg[4096])
+	char* msg)
 {
-	static  char	*id = "logon_pw";
-	int		i;
-	struct	passwd *pwdp;
-	SID	*usid;
-	char	*homedir;
-	char  	thepass[PWLEN+1];
-	char 	realname[UNLEN+1];
-	char	domain[PBS_MAXHOSTNAME+1] = "";
-	char	*p;
-	char	msg2[4096];
+	int		i = 0;
+	struct	passwd *pwdp; = NULL
+	SID	*usid = NULL;
+	char	*homedir = NULL;
+	char  	thepass[PWLEN+1] = {'\0'};
+	char 	realname[UNLEN+1] = {'\0'};
+	char	domain[PBS_MAXHOSTNAME+1] = {'\0'};
+	char	*p = {'\0'};
+	char	msg2[MSGLEN] = {'\0'};
```

Comment:
We don't need msg2 to act as an intermediate every time, but it's being used to "build" msg along the function a few times. Probably, we can do away with the use of msg2 when it's returning from the function in case of error. But in the rest of the places, it's being used for the log (line 4883, 4921).